### DISCUSSION TITLE
Design and structur of the export xlsx file

### DISCUSSION THREAD

**Discussion by provinzio** [provinzio/CoinTaxman#133] [General]
I tried my best to rework and improve the exported file, so that we can append it to our tax declaration without further worries.

I would be very interested to hear what you think of the new format. Do you have any ideas for improvement or better formulations?

@jhoogstraat @tobias-p

Edit: also checkout the `make archive` command/script to create a zip file with all relevant files for the evaluation.

**Comment by tobias-p** [comment]
First of all: Damn, that was a long ride to go... took me about 4 hours yesterday evening and today round about additional 8 hours... You should mention, how long the app needs for a specific number of lines to process so that a user can calculate, how long this have to run...

Second: The output xlsx file is pretty nice! But also, in my case it's huge as f*ck...
Verkauf = 1.313
Coin-Lending = 15.107
Staking = 735
Zinsen = 441
Airdrops = 549
Ein- und Auszahlungen = 103
All-in-All about 18.248 lines... 

My first thought: 🤯... My second: even if the "Finanzamt" wants to be more digital, there definitely will be someone who wants to print this out which leads to at least one dead tree.

So my idea would be: If possible, add another "shortened" version of "Verkauf", "Coin-Lending Einkünfte", "Staking Einkünfte", "Zinsen" and "Airdrops" where you just summarize e.g. the whole "Verkauf" of one coin for the whole year. This would make the whole xlsx file larger, but easier to handle for "alteingesessene Finanzbeamte", who want to print out something.

Also I would add a sheet for each exchange / wallet so the user can easily check out how much he owned each year / wallet...

**Comment by jhoogstraat** [comment]
The xlsx is amazing! The structure is very clear and organized.
In the "Verkauf" worksheet: Why are some column duplicated (annotated with (1) and (2)).
Everything with a (2) is 0 for me.

I do have unrealized gains, but the "Zusammenfassung" worksheet shows 0, is that correct?

**Reply by provinzio** [reply]
Both these points can lead to the fact, that your listed Verkauf has multiple different fee coins. Therefore multiple columns are required. 
To avoid having an unlimited number of columns or grouping all fees together, I added only two columns (in my case, it's never more than two).

1. CoinTaxman is currently implemented, so that operations at the same time and UTC time are grouped together
2. Binance offers you to pay your fees in BNB. If you haven't enough BNB, you have to pay the rest in the buying coin.

**Reply by jhoogstraat** [reply]
I definitely used BNB for fees at some point. I guess it has something to do with #136, which I am working on currently.
